static/src/js/components/PlatformField/PlatformField.jsx

```diff
@@ -0,0 +1,259 @@
+      showMenu: true,
+      shouldFocus: true
+    }))
+  }
+
+  const onHandleBlur = () => {
+    setState((prevState) => ({
+      ...prevState,
+      showMenu: false,
+      shouldFocus: false
+    }))
+  }
+
+  const onHandleClear = () => {
+    const { onChange } = props
+    setState((prevState) => ({
+      ...prevState,
+      type: '',
+      shortName: '',
+      longName: '',
+      showMenu: false,
+      shouldFocus: false
+    }))
+
+    onChange(state)
+  }
+
+  const displayTitle = (value) => (
+    <div className="platform-field-select-title">
+      {value}
+    </div>
+  )
+
+  const displaySelectOption = (value) => (
+    <button
+      type="button"
+      className="platform-field-select-option"
+      onMouseDown={() => onHandleMouseDown(value)}
+    >
+      {value[1]}
+    </button>
+  )
+
+  const displayClearOption = () => (
+    <button
+      type="button"
+      className="platform-field-clear-option"
+      onClick={onHandleClear}
+    >
+      Clear Short Name
+    </button>
+  )
+
+  const existingValue = {
+    value: state.shortName,
+    label: state.shortName
+  }
+
+  const selectOptions = [{
+    value: '',
+    label: ''
+  }, ...state.keyword.map((currentEnum) => ({
+    value: currentEnum,
+    label: currentEnum
+  }))]
+
+  // This is using the react-select components={{ Option }}, this allows us to
+  // render a custom dropdown. For the case of Short Name, we need hierarchical structure
+  // where it has a title (in this case it's the Type) followed by all the Short Name in that Type
+  // For Example: JET
+  //               A340-600
+  //               NASA S-B VIKING
+  // This Option function will render the list of Types and Short Name in this hierarchy
+  // using the > delimited path that was created during useEffect()
+
+  const Option = ({ ...data }) => {
+    const paths = data.value.split('>')
+    const value = paths.splice(1)
+
+    if (paths[0] === '') {
+      return displayClearOption()
+    }
+
+    if (value.length > 2) {
+      return null
+    }
+
+    if (value.length === 1) {
+      return displayTitle(value[0])
+    }
+
+    return displaySelectOption(value)
+  }
+
+  return (
+    <div className="platform-field">
+      {/* Short Name field */}
+      Short Name
+      <i className="eui-icon eui-required-o required-icon" />
+      <div>
+        <Select
+          id="shortName"
+          key={`platform_short-name--${state.shouldFocus}`}
+          autoFocus={state.shouldFocus}
+          placeholder="Select Short Name"
+          options={selectOptions}
+          isLoading={state.loading}
+          components={{ Option }}
+          value={existingValue.value ? existingValue : null}
+          onFocus={onHandleFocus}
+          onBlur={onHandleBlur}
+          menuIsOpen={state.showMenu}
+        />
+      </div>
+
+      {/* Type field */}
+      <div style={{ marginTop: '8px' }}>
+        Type
+        <input
+          className="platForm-field-text-field"
```

Comment:
Follow the new scss className naming convention, look at CustomWidgetWrapper or ProgressSection as an example.